test/handlers/erc721/depositBurn.js

```diff
@@ -0,0 +1,97 @@
+            ERC721MintableContract.new().then(instance => ERC721MintableInstance1 = instance),
+            ERC721MintableContract.new().then(instance => ERC721MintableInstance2 = instance)
+        ])
+        
+        BridgeInstance = await BridgeContract.new(chainID, RelayerInstance.address, relayerThreshold);
+
+        resourceID1 = Ethers.utils.hexZeroPad((ERC721MintableInstance1.address + Ethers.utils.hexlify(chainID).substr(2)), 32);
+        resourceID2 = Ethers.utils.hexZeroPad((ERC721MintableInstance2.address + Ethers.utils.hexlify(chainID).substr(2)), 32);
+        initialResourceIDs = [resourceID1, resourceID2];
+        initialContractAddresses = [ERC721MintableInstance1.address, ERC721MintableInstance2.address];
+        burnableContractAddresses = [ERC721MintableInstance1.address]
+
+        await Promise.all([
+            ERC721HandlerContract.new(BridgeInstance.address, initialResourceIDs, initialContractAddresses, burnableContractAddresses).then(instance => ERC721HandlerInstance = instance),
+            ERC721MintableInstance1.mint(depositerAddress, tokenID)
+        ]);
+
+        await ERC721MintableInstance1.approve(ERC721HandlerInstance.address, tokenID, { from: depositerAddress });
+
+        depositData = '0x' +
+            resourceID1.substr(2) +
+            Ethers.utils.hexZeroPad(Ethers.utils.hexlify(tokenID), 32).substr(2) +    // Deposit Amount        (32 bytes)
+            Ethers.utils.hexZeroPad(Ethers.utils.hexlify(32), 32).substr(2) +               // len(recipientAddress) (32 bytes)
+            Ethers.utils.hexZeroPad(recipientAddress, 32).substr(2);                        // recipientAddress      (?? bytes)
+    });
+
+    it('[sanity] burnableContractAddresses should be marked true in _burnList', async () => {
+        for (const burnableAddress of burnableContractAddresses) {
+            const isBurnable = await ERC721HandlerInstance._burnList.call(burnableAddress);
+            assert.isTrue(isBurnable, "Contract wasn't successfully marked burnable");
+        }
+    });
+
+    it('[sanity] ERC721MintableInstance1 tokenID has been minted for depositerAddress', async () => {
+        const tokenOwner = await ERC721MintableInstance1.ownerOf(tokenID);
+        assert.strictEqual(tokenOwner, depositerAddress);
+    });
+
+    it('depositAmount of ERC721MintableInstance1 tokens should have been burned', async () => {
+        await BridgeInstance.deposit(
+            chainID,
+            ERC721HandlerInstance.address,
+            depositData,
+            { from: depositerAddress }
+        );
+
+        const handlerBalance = await ERC721MintableInstance1.balanceOf(ERC721HandlerInstance.address);
+        assert.strictEqual(handlerBalance.toNumber(), 0);
+
+        const depositerBalance = await ERC721MintableInstance1.balanceOf(depositerAddress);
+        assert.strictEqual(depositerBalance.toNumber(), 0);
+
+        // This would be the ideal test for verifying the token
```

Comment:
Any ideas why this may be happening would be welcomed. You can view the errors thrown by looking at past failed test, but they were:

```shell
1) Contract: ERC721Handler - [Deposit Burn ERC721]
       depositAmount of ERC721MintableInstance1 tokens should have been burned:
     AssertionError: Did not fail
      at fails (node_modules/truffle-assertions/index.js:161:9)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at Context.<anonymous> (test/handlers/ERC721Handler/depositBurn.js:84:9)
  2) Contract: ERC721Handler - [Deposit Burn ERC721]
       "after each" hook: after test for "depositAmount of ERC721MintableInstance1 tokens should have been burned":
     Error: Returned error: invalid argument 0: hex string without 0x prefix
      at Object.ErrorResponse (node_modules/truffle/build/webpack:/node_modules/web3-core-helpers/src/errors.js:29:1)
      at /home/travis/build/ChainSafe/chainbridge-solidity/node_modules/truffle/build/webpack:/node_modules/web3-core-requestmanager/src/index.js:140:1
      at /home/travis/build/ChainSafe/chainbridge-solidity/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:112:1
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/webpack:/node_modules/web3-providers-http/src/index.js:96:1)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:208:1)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:318:1)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:289:47)
      at endReadableNT (_stream_readable.js:1187:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Reply:
I would like to address this before merging, will look into it

Reply:
No immediate solutions, probably not a bug of ours. LGTM